src/session/components/message_entry.rs

```diff
@@ -154,6 +170,14 @@ impl MessageEntry {
         self.connect_notify_local(Some("formatted-text"), f)
     }
 
+    pub(crate) fn placeholder_text(&self) -> glib::GString {
+        self.imp().placeholder.text()
+    }
+
+    pub(crate) fn set_placeholder_text(&self, placeholder_text: &str) {
+        self.imp().placeholder.set_text(placeholder_text);
```

Comment:
Any reason you don't follow the usual scheme here by checking for equality and notifying?

Reply:
This is because here the `placeholder-text` property is acting as a wrapper to the label's `text` property, so it's the label's job to do the equality checking, but actually I forgot that this still needs a `notify` because otherwise if you call this method directly, the notify won't be called.